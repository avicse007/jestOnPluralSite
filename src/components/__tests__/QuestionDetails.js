import {mapStateToProps,QuestionDetailDisplay} from '../QuestionDetail.jsx';
import Renderer from "react-test-renderer";
import React from "react";


describe('The question details ',()=>{
    describe('The conatiner Element',()=>{
        describe("mapStateToProp",()=>{
            it("should map the state to prop correctly",()=>{
                const sampleQuestion = {
                    question_id: 42,
                    body: "Space is big"
                };
                const appState = {
                    questions:[sampleQuestion]
                };
                const ownProps = {
                        question_id:42
                };
                const componentState = mapStateToProps(appState,ownProps);
                console.log(componentState);
                expect(componentState).toEqual(sampleQuestion);
            });
        });
    });
    describe("The display Element ",()=>{
        it('should not regress',()=>{
            const tree = Renderer.create(
            <QuestionDetailDisplay
                title="Naming of life"
                body ="42"
                answer_count = {0}
                tags={["Hello"]}
            />);
            expect(tree.toJSON()).toMatchSnapshot();
        });

    });
   

});