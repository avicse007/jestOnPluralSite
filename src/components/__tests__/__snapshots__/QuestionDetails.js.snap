// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The question details  The display Element  should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Naming of life
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          Hello
        </code>
      </div>
    </div>
    <div>
      <p>
        42
      </p>
    </div>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
